src/allocator.rs

```diff
@@ -8,10 +8,11 @@ pub enum SExp {
     Pair(NodePtr, NodePtr),
 }
 
-#[derive(Clone, Copy)]
+// TODO: figure out how to integrate with RangeSet without exposing these fields
+#[derive(Clone, Copy, Debug)]
 pub struct AtomBuf {
-    start: u32,
-    end: u32,
+    pub start: u32,
```

Comment:
Consider a "getter" that returns these two values rather than making them public and giving people the wrong idea that they're mutable.

Reply:
Yeah, I think the main issue is that you can construct your own instances of this now, that's the main violation of the abstraction. I'll look into fixing this

Reply:
I restored this abstraction at a slight expense in complexity in `sanitize_integer`. See this commit: https://github.com/Chia-Network/clvm_rs/pull/95/commits/ccd8d073b48c1152ab57506ea2e3e57219a5ae03